pkg/client/cache/listers.go

```diff
@@ -151,11 +151,11 @@ type storeToNodeConditionLister struct {
 }
 
 // List returns a list of nodes that match the conditions defined by the predicate functions in the storeToNodeConditionLister.
-func (s storeToNodeConditionLister) List() (nodes api.NodeList, err error) {
+func (s storeToNodeConditionLister) List() (nodes []*api.Node, err error) {
```

Comment:
@lavalamp @davidopp 

This is the main change in this PR (most of other change are test fixes and stuff like that).

This has significant performance impact on scheduler, but before I finish fixing tests, I would like to get your opinion on it.

In particular, I think that:
- ideally api.XxxList should all be []*api.Xxx instead of []api.Xxx (e.g. api.NodeList should be []*api.Node instead of []api.Node (plus all the metadata that we currently have))
- however, since this seems to be backward-incompatible change, we may consider changing all listers here to api like I'm changing it here.

Thoughts?